core-api/src/main/java/com/optimizely/ab/config/audience/UserAttribute.java

```diff
@@ -76,14 +81,40 @@ public Boolean evaluate(ProjectConfig config, Map<String, ?> attributes) {
         Object userAttributeValue = attributes.get(name);
 
         if (!"custom_attribute".equals(type)) {
-            MatchType.logger.error(String.format("condition type not equal to `custom_attribute` %s", type));
+            logger.warn("Audience condition \"{}\" has an unknown condition type. You may need to upgrade to a newer release of the Optimizely SDK", this.toString());
```

Comment:
calling `this.toString()` is unnecessary as the logger will do this if you just pass `this`, but it will do it lazily in case `WARN` messages are not output

Reply:
Ooh, yeah, that makes sense! +1 for only passing `this`